blockchain/pending.go

```diff
@@ -65,3 +67,22 @@ func (p *PendingState) Class(classHash *felt.Felt) (*core.DeclaredClass, error)
 
 	return p.head.Class(classHash)
 }
+
+// Note[pnowosie]: Maybe extending StateReader with the following methods was not a good idea?
+func (s *PendingState) ClassTrie() (*trie.Trie, func() error, error) {
+	return nil, nopCloser, featureNotImplemented
+}
+func (s *PendingState) StorageTrie() (*trie.Trie, func() error, error) {
+	return nil, nopCloser, featureNotImplemented
+}
+func (s *PendingState) StorageTrieForAddr(*felt.Felt) (*trie.Trie, error) {
+	return nil, featureNotImplemented
+}
+func (s *PendingState) StateAndClassRoot() (*felt.Felt, *felt.Felt, error) {
+	return nil, nil, featureNotImplemented
+}
```

Comment:
It looks like a bad idea to extend a StateReader which is implemented by other state variants like StateHistoryReader.
Otherwise - for tests we mock StateHistoryReader so in case of separate interface testing might be a bit harsh - ideas?